src/kOS/Suffixed/VesselTarget.cs

```diff
@@ -470,6 +473,16 @@ private void InitializeSuffixes()
             return crew;
         }
 
+        public MessageQueueStructure GetMessages()
+        {
+            if (Shared.Vessel.id != Vessel.id)
+            {
+                throw new KOSException("You can only access the message queue of the current vessel");
```

Comment:
Yeah, there could be a benefit - someone might want to catch only `KOSCommunicationException`s. We don't have exception catching in kerboscript for now so I wasn't concentrating on this. I'll have a look at it again later, I'll add a new exception if it makes sense.